backend/src/routes/api/notebooks/index.ts

```diff
@@ -49,7 +51,22 @@ module.exports = async (fastify: KubeFastifyInstance) => {
           ? false
           : await getNotebookStatus(fastify, namespace, name);
 
-        return { notebook, isRunning };
+        const notebookName = notebook?.metadata.name;
+        let newNotebook: Notebook | void;
+        if (isRunning && !notebook?.metadata.annotations?.['opendatahub.io/link']) {
+          const route = await getRoute(fastify, namespace, notebookName).catch((e) => {
+            fastify.log.warn(`Failed patching route to notebook ${notebookName}, ${e.message}`);
+          });
+          if (route) {
```

Comment:
nit: this could be written as a guard:
```
if (!route) {
  // already issued the warning above
  return;
}
```

Reply:
Disagree in the current format of the code. `return` has a lot of status updates... we'd need to move this logic to a function to exit early.